SofaKernel/modules/SofaBaseLinearSolver/CGLinearSolver.inl

```diff
@@ -152,21 +158,24 @@ void CGLinearSolver<TMatrix,TVector>::solve(Matrix& M, Vector& x, Vector& b)
             simulation::Visitor::printNode(comment.str());
         }
 #endif
-        // 		printWithElapsedTime( x, helper::system::thread::CTime::getTime()-time0,sout );
 
-        //z = r; // no precond
-        //rho = r.dot(z);
+
+        /// Compute ρ = r²
         rho = r.dot(r);
 
+        /// If NOT the first step
         if (nb_iter>1)
         {
-            double normr = sqrt(rho); //sqrt(r.dot(r));
+            /// Compute the error from the norm of ρ and b
+            double normr = sqrt(rho);
             double err = normr/normb;
+
             graph_error.push_back(err);
+
+            /// Break condition = TOLERANCE criterion regarding the error err is reached
             if (err <= f_tolerance.getValue())
             {
                 endcond = "tolerance";
-
 #ifdef SOFA_DUMP_VISITOR_INFO
```

Comment:
Hi hugo, 

If it is possible I suggest you to convert the #ifdef pattern with this one:
```cpp
#ifdef SOFA_DUMP_VISITOR_INFO
#define DO_DUMP_VISITOR_INFO true
#else
#define DO_DUMP_VISITOR_INFO false
#endif 
```

Then in the code you can use a real if:
```cpp
if(DO_DUMP_VISITOR_INFO){
    .... 
} 
```
This pattern allow the conditional code to be compiled each time so it ease the CI as well as doing automated refactoring (as offered by qtcreator).  

In term of performance I assume that a modern compiler should totally remove the dead-code according to the true/false condition.

Reply:
Hello Hugo, 
Thank for having tried...and for the DISPLAY_TIME one ? 

More generally (out of the scope of this PR) what would be the cost of removing the #ifdef *DUMP* in the simulation::Visitor signature ?